Apps/phcheckpointfirewall/checkpoint_connector.py

```diff
@@ -486,6 +487,45 @@ def _unblock_ip(self, param):
 
         return action_result.set_status(phantom.APP_SUCCESS, "Successfully unblocked {0}".format('subnet' if length != '32' else 'IP'))
 
+    def _add_host(self, param):
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        if not(self._login(action_result)):
+            return action_result.get_status()
+
+        ip = param.get(phantom.APP_JSON_IP, None)
+        ipv4 = param.get('ipv4', None)
+        ipv6 = param.get('ipv6', None)
+        name = param.get('name', None)
+
+        endpoint = 'add-host'
+
+        if ip:
+            ret_val, resp_json = self._make_rest_call(endpoint, {'name': name, 'ip-address': ip}, action_result)
+        elif ipv4 and ipv6:
+            ret_val, resp_json = self._make_rest_call(endpoint, {'name': name, 'ipv4-address': ipv4, 'ipv6-address': ipv6}, action_result)
+        elif ipv4:
+            ret_val, resp_json = self._make_rest_call(endpoint, {'name': name, 'ipv4-address': ipv4}, action_result)
+        elif ipv6:
+            ret_val, resp_json = self._make_rest_call(endpoint, {'name': name, 'ipv6-address': ipv6}, action_result)
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "You must specify an ip address")
+
+        if ((not ret_val) and (not resp_json)):
```

Comment:
nit:
```suggestion
        if (not ret_val) and (not resp_json):
```

redundant parentheses are generally discouraged in python styling. this is the case for parentheses around keyword components like `if`, `while`, etc. (unless you're trying to create a multiline condition)

you also technically don't need parentheses around the inner expressions in these cases albeit having those or not is more of a style preference whether or not you think it's more readable. the order of operations will still be correct without them

ditto for later cases of the same (won't duplicate this comment)

Reply:
changes incorporated in d2e6a92